`TestJaegerGRPCMuxed*` system tests are flaky

Seen in a few PRs lately, it may be a result of recent refactoring efforts:

```
 === RUN   TestJaegerGRPCMuxed
    jaeger_test.go:71: 
        	Error Trace:	/var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9911/src/github.com/elastic/apm-server/systemtest/jaeger_test.go:71
        	            				/var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9911/src/github.com/elastic/apm-server/systemtest/jaeger_test.go:47
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestJaegerGRPCMuxed
    server.go:132: log file: /var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9911/src/github.com/elastic/apm-server/systemtest/logs/TestJaegerGRPCMuxed/apm-server
--- FAIL: TestJaegerGRPCMuxed (2.29s)
```

```
=== RUN   TestJaegerGRPCMuxedTLS
    jaeger_test.go:71: 
        	Error Trace:	/var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9911/src/github.com/elastic/apm-server/systemtest/jaeger_test.go:71
        	            				/var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9911/src/github.com/elastic/apm-server/systemtest/jaeger_test.go:56
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestJaegerGRPCMuxedTLS
    server.go:132: log file: /var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9911/src/github.com/elastic/apm-server/systemtest/logs/TestJaegerGRPCMuxedTLS/apm-server
--- FAIL: TestJaegerGRPCMuxedTLS (1.95s)
```

### Related PRs

https://github.com/elastic/apm-server/pull/9911#issuecomment-1366547405

--- a/systemtest/aggregation_test.go
+++ b/systemtest/aggregation_test.go
@@ -67,16 +67,17 @@ func TestTransactionAggregation(t *testing.T) {
 	systemtest.SendBackendEventsLiteral(t, srv.URL, faasPayload)
 	tracer.Flush(nil)
 
-	// Make sure apm-server.aggregation.txmetrics metrics are published. Metric values are unit tested.
-	doc := getBeatsMonitoringStats(t, srv, nil)
-	assert.True(t, gjson.GetBytes(doc.RawSource, "beats_stats.metrics.apm-server.aggregation.txmetrics").Exists())
-
 	// Wait for the transaction to be indexed, indicating that Elasticsearch
 	// indices have been setup and we should not risk triggering the shutdown
 	// timeout while waiting for the aggregated metrics to be indexed.
 	systemtest.Elasticsearch.ExpectDocs(t, "traces-apm*",
 		estest.TermQuery{Field: "processor.event", Value: "transaction"},
 	)
+
+	// Make sure apm-server.aggregation.txmetrics metrics are published. Metric values are unit tested.
+	doc := getBeatsMonitoringStats(t, srv, nil)
+	assert.True(t, gjson.GetBytes(doc.RawSource, "beats_stats.metrics.apm-server.aggregation.txmetrics").Exists())
+
 	// Stop server to ensure metrics are flushed on shutdown.
 	assert.NoError(t, srv.Close())
 

--- a/systemtest/jaeger_test.go
+++ b/systemtest/jaeger_test.go
@@ -67,14 +67,13 @@ func testJaegerGRPC(t *testing.T, srv *apmservertest.Server, addr string, dialOp
 	_, err = client.PostSpans(context.Background(), request)
 	require.NoError(t, err)
 
-	doc := getBeatsMonitoringStats(t, srv, nil)
-	assert.Equal(t, int64(1), gjson.GetBytes(doc.RawSource, "beats_stats.metrics.apm-server.jaeger.grpc.collect.request.count").Int())
-
 	systemtest.Elasticsearch.ExpectDocs(t, "traces-apm*", estest.BoolQuery{Filter: []interface{}{
 		estest.TermQuery{Field: "processor.event", Value: "transaction"},
 	}})
-
 	// TODO(axw) check document contents. We currently do this in beater/jaeger.
+
+	doc := getBeatsMonitoringStats(t, srv, nil)
+	assert.Equal(t, int64(1), gjson.GetBytes(doc.RawSource, "beats_stats.metrics.apm-server.jaeger.grpc.collect.request.count").Int())
 }
 
 func TestJaegerGRPCSampling(t *testing.T) {

